dfm/description/matchcounter.py

```diff
@@ -0,0 +1,129 @@
+from collections import defaultdict
+
+from spacy.language import Language
+from spacy.matcher import Matcher
+from spacy.tokens import Doc
+
+from typing import Dict, Iterable, List, Optional
+
+
+class MatchCounter:
+    """Class for counting matches in a text corpus.
+
+    Args:
+        match_patterns (List[Dict[str, list]): list of lowercase spacy match patterns
+        nlp (Language): The spacy language to use
+    """
+
+    def __init__(self, match_patterns: List[Dict[str, list]], nlp: Language):
+        self.nlp = nlp
+        self.matcher_objects = self.gen_matcher_objects_from_pattern_list(
+            match_patterns
+        )
+
+    @staticmethod
+    def term_list_to_lowercase_match_patterns(
+        term_list: List[str],
+        label_prefix: Optional[str] = "",
+        label: Optional[str] = None,
+    ) -> List[str]:
+        """Takes a list of strings and converts it to a list of spacy match patterns
+
+        Args:
+            term_list (List[str]): List of terms.
+            label_prefix (Optional[str], optional): Prefix for the label. Helpful when aggregating.
+                E.g. you want your occupations to be prefixed with "occu_" for later processing, like "occu_nurse", "occu_doctor" etc.
+                Defaults to "".
+            label (Optional[str], optional): Label for the spacy match patterns. Helpful when aggregating, e.g. you'd like to count all your male names in a "male names" variable.
+                Defaults to None, indicating each pattern will be labeled by its term.
+
+        Returns:
+            List[str]: Spacy match patterns in the shape {"label": [{"LOWER": "term"}]}
+        """
+        out_list = []
+
+        for term in term_list:
+            if label is None:
+                cur_label = label_prefix + term
+                out_list.append({cur_label: [{"LOWER": term}]})
+            else:
+                cur_label = label_prefix + label
+                out_list.append({cur_label: [{"LOWER": term}]})
+
+        return out_list
+
+    def gen_matcher_objects_from_pattern_list(
```

Comment:
```suggestion
    def generates_matcher_objects_from_pattern_list(
```
Abbreviations are bad, non standard abbreviations are worse :-) https://gist.github.com/klaaspieter/12cd68f54bb71a3940eae5cdd4ea1764